src/main/kotlin/org/jitsi/nlj/transform/node/incoming/VideoParser.kt

```diff
@@ -61,51 +67,78 @@ class VideoParser(
                     val vp8Packet = packetInfo.packet.toOtherType(::Vp8Packet)
                     packetInfo.packet = vp8Packet
                     packetInfo.resetPayloadVerification()
+
+                    if (videoCodecParser !is Vp8Parser) {
+                        resetSources()
+                        videoCodecParser = Vp8Parser(sources, logger)
+                    }
+                }
+                is Vp9PayloadType -> {
+                    val vp9Packet = packetInfo.packet.toOtherType(::Vp9Packet)
+                    packetInfo.packet = vp9Packet
+                    packetInfo.resetPayloadVerification()
+
+                    if (videoCodecParser !is Vp9Parser) {
+                        resetSources()
+                        videoCodecParser = Vp9Parser(sources, logger)
+                    }
+                }
+                else -> {
+                    videoCodecParser = null
                 }
             }
+            videoCodecParser?.parse(packetInfo)
+            packet = packetInfo.packetAs()
         } catch (e: Exception) {
             logger.error("Exception parsing video packet.  Packet data is: " +
-                "${packet.buffer.toHex(packet.offset, Math.min(packet.length, 80))}", e)
+                packet.buffer.toHex(packet.offset, Math.min(packet.length, 80)), e)
             return null
         }
 
-        val videoPacket = packetInfo.packetAs<VideoRtpPacket>()
-        val encodingDesc = findRtpLayerDesc(videoPacket) ?: run {
-            logger.warn("Unable to find encoding matching packet! packet=$videoPacket, " +
-                "encodings=${sources.joinToString(separator = "\n", limit = 1, truncated = "[${sources.size - 1} " +
-                    "more source descriptions omitted]")}")
-            numPacketsDroppedNoEncoding.incrementAndGet()
-            return null
+        if (packet is ParsedVideoPacket) {
```

Comment:
This will only happen if we don't recognize/support the PT, right?  Might be better to do an early return in that case and we can know that once we get out of that block we've got valid data (and can get rid of this check)